packages/menu/src/SubMenu.tsx

```diff
@@ -0,0 +1,349 @@
+    } else if (onClick) {
+      onClick(e);
+    }
+  };
+
+  const renderedSubMenuItem = (Root: React.ElementType<any> = 'button') => (
+    <li role="none" className={liStyle}>
+      <Root
+        {...subMenuContainer.prop}
+        {...rest}
+        onKeyDown={onKeyDown}
+        role="menuitem"
+        href={href}
+        aria-haspopup="true"
+        ref={ref as RefObject<any>}
+        onClick={onRootClick}
+        className={cx(
+          menuItemContainerStyle,
+          subMenuStyle,
+          linkStyle,
+          {
+            [activeMenuItemContainerStyle]: active,
+            [disabledMenuItemContainerStyle]: disabled,
+            [subMenuOpenStyle]: open,
+            [focusedMenuItemContainerStyle]: showFocus,
+          },
+          className,
+        )}
+      >
+        {glyph && (
+          <Icon
+            glyph={glyph}
+            size="xlarge"
+            className={cx(mainIconStyle, {
+              [activeIconStyle]: active,
+              [mainIconFocusedStyle]: showFocus,
+            })}
+          />
+        )}
+
+        <div>
+          <div
+            className={cx(mainTextStyle, {
+              [activeMainTextStyle]: active,
+            })}
+          >
+            {title}
+          </div>
+
+          <div className={cx({ [disabledTextStyle]: disabled })}>
+            {description}
+          </div>
+        </div>
+      </Root>
+      <IconButton
+        ref={iconButtonRef}
+        ariaLabel={open ? 'Caret Up' : 'Caret Down'}
+        className={cx(iconButtonStyle, {
+          [openIconButtonStyle]: open,
+          [iconButtonFocusedStyle]: showFocus,
+        })}
+        onClick={(e: React.MouseEvent) => {
+          e.nativeEvent.stopImmediatePropagation();
+
+          if (setOpen) {
+            setOpen(!open);
+          }
+        }}
+      >
+        <Icon
+          glyph={open ? 'CaretUp' : 'CaretDown'}
+          className={open ? openIconStyle : closedIconStyle}
+        />
+      </IconButton>
+      <Transition
+        in={open}
+        timeout={{
+          enter: 0,
+          exit: 150,
+        }}
+        mountOnEnter
+        unmountOnExit
+        onExited={onExited}
+      >
+        {(state: string) => (
+          <ul
+            className={cx(ulStyle, {
+              [css`
+                height: ${subMenuItemHeight *
+                  React.Children.toArray(children).length}px;
+              `]: state === 'entered',
+            })}
+            role="menu"
+            aria-label={title}
+          >
+            {React.Children.map(children as React.ReactElement, child => {
+              return React.cloneElement(child, {
+                children: (
+                  <>
+                    <div className={menuItemBorder}></div>
+                    {child.props.children}
+                  </>
+                ),
+                className: cx(
+                  css`
+                    padding-left: ${glyph ? '60px' : '28px'};
```

Comment:
I don't understand from looking at this why those values are used

Reply:
👍